scripts/npx-helper.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+#
+# Runs the given command with npx, ensuring that the version of the
```

Comment:
Not following the purpose of this script.

Reply:
In you pull this PR, and run `make spellcheck`, you'll get the same behavior as before. That is, `cspell` will be installed (since you haven't run a full `npm install` yet).

Generally, if you run `npx cspell ...`, it will install the latest version. For the sake of consistency in CI and local environments, it is best to use the same version of packages. The `npx-helper.sh` script ensures that by fetching the same version of that package as specified in the `package.json` file.

Reply:
if we're standardizing on `npm install` being a prerequisite to building anything, then `make spellcheck` should simply fail if `npm install` wasn't done.

And to guarantee completely repeatable builds, why not check in the lock file at the root? We had it previously committed as scripts/cspell/package-lock.json

Reply:
> if we're standardizing on `npm install` being a prerequisite to building anything,

Sure, that is what I prefer as well. I was just trying to preserve the old behavior (something that the OTel spec repo folks wanted).

Then I can drop the `npx-helper.sh`.

Reply:
> And to guarantee completely repeatable builds, why not check in the lock file at the root? We had it previously committed as scripts/cspell/package-lock.json

This is a website build, without any fancy JS / Single-page apps etc. I've pinned to specific versions in the `package.json` file where necessary, so we don't need the lock file here IMHO. I've omitted it here, and for websites like OTel.io, because I've found it a pain (on other projects) to manage where there are a lot of community contributions, especially when their PRs stay open for a long time. We then have to deal with lock file merge conflicts.

But it's your call. Let me know what you prefer.